generators/binarygen/analysis/tags.go

```diff
@@ -0,0 +1,241 @@
+	case "AtStart":
+		out.subsliceStart = AtStart
+	case "AtCurrent":
+		out.subsliceStart = AtCurrent
+	case "":
+		// make AtStart the default for opaque types
+		if out.isOpaque {
+			out.subsliceStart = AtStart
+		} else {
+			out.subsliceStart = AtCurrent
+		}
+	default:
+		panic("invalic tag for subsliceStart : " + tag)
+	}
+
+	switch tag := tags.Get("arrayCount"); tag {
+	case "FirstUint16":
+		out.arrayCount = FirstUint16
+	case "FirstUint32":
+		out.arrayCount = FirstUint32
+	case "ToEnd":
+		out.arrayCount = ToEnd
+	default:
+		if _, field, hasComputedField := strings.Cut(tag, "ComputedField-"); hasComputedField {
+			out.arrayCount = ComputedField
+			out.arrayCountField = field
+		} else if _, field, hasToField := strings.Cut(tag, "To-"); hasToField {
+			out.arrayCount = ToComputedField
+			out.arrayCountField = field
+		} else if tag == "" {
+			// default to NoLength
+			out.arrayCount = NoLength
+		} else {
+			panic("invalid tag for arrayCount: " + tag)
+		}
+	}
+
+	switch tag := tags.Get("offsetSize"); tag {
+	case "Offset16":
+		out.offsetSize = Offset16
+	case "Offset32":
+		out.offsetSize = Offset32
+	case "":
+	default:
+		panic("invalid tag for offsetSize: " + tag)
+	}
+
+	switch tag := tags.Get("offsetsArray"); tag {
+	case "Offset16":
+		out.offsetsArray = Offset16
+	case "Offset32":
+		out.offsetsArray = Offset32
+	case "":
+	default:
+		panic("invalid tag for offsetsArray: " + tag)
+	}
+
+	switch tag := tags.Get("offsetRelativeTo"); tag {
+	case "Parent":
+		out.offsetRelativeTo = Parent
+	case "GrandParent":
+		out.offsetRelativeTo = GrandParent
+	case "":
+	default:
+		panic("invalid tag for offsetRelativeTo: " + tag)
+	}
+
+	unionField := tags.Get("unionField")
+	if unionField != "" {
+		for i := 0; i < st.NumFields(); i++ {
+			if fi := st.Field(i); fi.Name() == unionField {
+				out.unionField = fi
+				break
+			}
+		}
+		if out.unionField == nil {
+			panic("unknow field for union version: " + unionField)
+		}
+	}
+
+	if unionTag := tags.Get("unionTag"); unionTag != "" {
+		value, err := strconv.Atoi(unionTag)
+		if err != nil {
+			panic(err)
+		}
+		out.unionTag = constant.MakeInt64(int64(value))
+	}
+
+	if args := tags.Get("arguments"); args != "" {
+		chunks := strings.Split(tags.Get("arguments"), ",")
+
+		for _, chunk := range chunks {
+			forName, value, ok := strings.Cut(chunk, "=")
+			if !ok {
+				panic("expected <argName>=<value>, got " + chunk)
+			}
+			out.requiredFieldArguments = append(out.requiredFieldArguments, ProvidedArgument{
+				Value: strings.TrimSpace(value),
+				For:   strings.TrimSpace(forName),
+			})
+		}
+	}
+
+	return out
+}
+
+type Argument struct {
+	VariableName string
+	TypeName     string
+}
+
+type commments struct {
+	// externalArguments may be provided it the type parsing/writting function
+	// requires data not provided in the input slice
+	externalArguments []Argument
+}
+
+// parse the type documentation looking for special comments
+// of the following form :
```

Comment:
Looks like this comment is missing an example comment that demonstrates the formatting.

Reply:
Indeed, thank you. Fixed.